shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -344,27 +352,32 @@ namespace datadog::shared::nativeloader
               {
                   return std::nullopt;
               }
-              if (eval_result == plcs::Result::TRUE)
-              {
-                  injection_status = InjectionStatus::ALLOW;
-              }
-              else
-              {
-                  injection_status = InjectionStatus::DENY;
-              }
 
+              injection_status = (eval_result == plcs::Result::TRUE) ? InjectionStatus::ALLOW : InjectionStatus::UNKNOWN;
               return std::nullopt;
           });
 
-          const auto wls_file = GetPoliciesPath();
-          if (fs::exists(wls_file))
-          {
-            auto maybe_error = plcs::evaluate_buffer_from_file(wls_file);
-            if (maybe_error) {
-              Log::Error("CorProfiler::Initialize: An error occured while evaluating workload selection (reason: ", *maybe_error, ")");
-              return E_FAIL;
+          plcs::register_action(plcs::Action::INJECT_DENY, [&injection_status](plcs::Result eval_result, const std::vector<const char*>&, const char* desc) -> std::optional<plcs::Error> {
+            if (injection_status == InjectionStatus::DENY)
```

Comment:
We should update the status only if it's unknown, the first rule that fires a Deny or an Allow wins.